cupy/random/_generator_api.pyx

```diff
@@ -159,6 +159,75 @@ class Generator:
         # omitted args.
         return (<object>y).astype(dtype, copy=False)
 
+    def uniform(self, low=0.0, high=1.0, size=None, dtype=numpy.float64):
+        """
+        Draw samples from a uniform distribution.
+        Samples are uniformly distributed over the half-open interval
+        ``[low, high)`` (includes low, but excludes high).  In other words,
+        any value within the given interval is equally likely to be drawn
+        by `uniform`.
+
+        Parameters
+        ----------
+        low : float or array_like of floats, optional
+            Lower boundary of the output interval.  All values generated will
+            be greater than or equal to low.  The default value is 0.
+        high : float or array_like of floats
+            Upper boundary of the output interval.  All values generated will
+            be less than high.  The high limit may be included in the returned
+            array of floats due to floating-point rounding in the equation
+            ``low + (high-low) * random()``.  high - low must be
+            non-negative.  The default value is 1.0.
+        size : int or tuple of ints, optional
+            Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
+            ``m * n * k`` samples are drawn.  If size is ``None`` (default),
+            a single value is returned if ``low`` and ``high`` are both
+            scalars.  Otherwise, ``cupy.broadcast(low, high).size`` samples are
+            drawn.
+
+        Returns
+        -------
+        out : ndarray or scalar
+            Drawn samples from the parameterized uniform distribution.
+        See Also
+        --------
+        - :meth:`numpy.random.Generator.uniform`
+        - :meth:`integers`: Discrete uniform distribution, yielding integers.
+        - :meth:`random`: Floats uniformly distributed over ``[0, 1)``.
+        """
+
+        cdef _ndarray_base y
+
+        if not isinstance(low, _ndarray_base):
+            if type(low) in (float, int):
+                low = cupy.asarray(low, numpy.float64)
+            else:
+                raise TypeError('low is required to be a cupy.ndarray'
+                                ' or a scalar')
+        if not isinstance(high, _ndarray_base):
+            if type(high) in (float, int):
+                high = cupy.asarray(high, numpy.float64)
+            else:
+                raise TypeError('high is required to be a cupy.ndarray'
+                                ' or a scalar')
+
+        if size is not None and not isinstance(size, tuple):
+            size = (size, )
+        elif size is None:
+            size = cupy.broadcast(low, high).shape
```

Comment:
`_core.ndarray()` accepts scalar integer type as `size`.
```suggestion
        if size is None:
            size = cupy.broadcast(low, high).shape
```

Reply:
Great, I made this change.  But do you know why the original numpy code doesn't do this?  numpy's ndarray also accepts scalar integer types as size.

Reply:
Not sure but maybe because of historical reasons :thinking: